unit7/lesson3/U7L3Q2/U7_L3_Activity_Two.java

```diff
@@ -3,54 +3,43 @@
 public class U7_L3_Activity_Two
 {
 
-  // Write the printStatistics method as described in the assignment
-
-public static void printStatistics(ArrayList<Integer> list) {
-  
-  int sum = 0;
-  double avg = 0;
-  int mode = 0;
-  double maxValue = Double.NaN;
-  int counter = 0;
-  int maxCount = 0;
-  boolean infinite = false;
-  ArrayList<Integer> hold = new ArrayList<Integer>();
-  
-  
-  for (int i = 0; i < list.size(); i++) {
-  sum += list.get(i); 
-  counter++;
-  avg += list.get(i);
-} 
-  avg = avg / counter;
-  
-  for (int i = 0; i < list.size(); i++) {
-    int count = 0;
-    for (int j = i; j < list.size(); j++) {
-      if (list.get(j) == list.get(i)) 
-        count++;
-    }
-        if (count == maxCount) {
-            maxValue = Double.NaN;
-            infinite = true;
-        }
-        else if (count > maxCount) {
-            maxCount = count;            
-            maxValue = list.get(i);;
+    // Write the printStatistics method as described in the assignment
+    public static void printStatistics(ArrayList<Integer> nums) {
+      // define variables at start
+      int sum = 0;
+      int mode = 0;
+      int modeRec = 0; // records the "world-record" number of times a num repeats
+      int num;
+      boolean validMode = false; // used to check if the mode is valid
+      // loop through nums
+      for (int i = 0; i < nums.size(); i++) {
+        // universal variable for what number the outer loop is "on"
+        num = nums.get(i);
+        sum += num;
+        // define modeCount to track the current frequency of num, then discard when num is re-initialized
+        int modeCount = 0;
+        // inner loop to compare num against all integers in nums
+        for (int k = 0; k < nums.size(); k++) {
+          // adds to loop count if a repeat is found
+          if (num == nums.get(k) && (num != mode || modeRec == 0)) {
+            modeCount++;
           }
+        }
+        // determines if current count is higher then "world-record"
+        if (modeCount > modeRec) {
+          modeRec = modeCount;
+          mode = num;
+          validMode = true;
+          // if the current count equals the "world-record", temporarily set validMode to false until new mode is found or loop finishes
+        } else if (modeCount == modeRec) {
+          validMode = false;
+        }
       }
-  
-  
-  System.out.println("Sum: " + sum);
-  System.out.println("Average: " + avg);
-  if(infinite == true)
-  System.out.println("Mode: no single mode");
-  else if(maxCount > 1){
-  mode = (int) maxValue;
-  System.out.println("Mode: " + mode);
-}
-  else 
-  System.out.println("Mode: no single mode");
-}
-
+      System.out.print("Sum: " + sum + "\nAverage: " + ((double) sum/nums.size()) + "\nMode: ");
+      if (validMode) {
+        System.out.println(mode);
+      } else {
+        System.out.println("no single mode");
+      }
```

Comment:
A ternary operator (short hand if/else statement) can be used here to further simplify the code.

```suggestion
      // Ternary operator - (condition ? if-true : if-false)
      System.out.println(validMode ? mode : "no single mode");
```
So...
- if validMode = true, print `mode`.
- if validMode = false, print `"no single mode"`.

Reply:
Apparently not many people know about this... so up to you whether you want to replace it or not.